tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -420,6 +420,139 @@ def verify_failure_received():
         finally:
             lambda_client.delete_event_source_mapping(UUID=event_source_mapping_uuid)
 
+    @pytest.mark.parametrize(
+        "item_to_put1, item_to_put2, filter, calls",
+        [
+            # Test with no filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                None,
+                2,
+            ),
+            # Test with filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                {"eventName": ["INSERT"]},
+                1,
+            ),
+            # Test with OR filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
```

Comment:
When running the test against AWS, the second insert wasn't recognized as a "modify" event. When I changed the second insert to `{"id": {"S": "test123"},"test": {"S": "modified"}}` it would work. 

There might be some hidden logic in AWS that prevents firing the "modify" when the item did not change at all. Please adapt the test cases accordingly.